### ISSUE TITLE
ISPC produces wrong code with bool type iterator.

### ISSUE THREAD

**Issue by Vsevolod-Livinskij** [ispc/ispc#1806]
The loop should be execuded once while ISPC skips it. This happens on a lot of different targets.
> \>$ ispc func.ispc -o func.o && clang++ func.o driver.cpp && ./a.out
BAD

Reproducer:
func.ispc
```c++
export void test() {
    for (uniform bool i = 0; i < (uniform bool)1; i += 1)
    {
        print("OK\n");
        return;
    }
    print("BAD\n");
}
```

driver.cpp
```c++
extern "C" { void test(); }

int main() {
    test();
}
```

ISPC version:
1.14.0dev (build commit 1bc127f35c0d0780 @ 20200623, LLVM 10.0.0) + da679386632e25bfb9d82de7f4cd9b9d88e8c937

**Comment by dbabokin** [comment]
```c
i < (uniform bool)1
``` 
compare triggers the following LLVM IR: 
```llvm
%less_i_load_trunc_ = icmp slt i1 %i_load_trunc, true
```

Note comparison type ``slt``, which mean that i1 is signed type. I.e. ``0`` is treated as ``0`` and ``1`` is treated as ``-1``. This is incorrect and comparison type should be ``ult``.

**Comment by dbabokin** [comment]
Fixed by #1811
